pandas/core/internals.py

```diff
@@ -1837,8 +1837,9 @@ def _can_hold_element(self, element):
         if tipo is not None:
             return (issubclass(tipo.type, (np.floating, np.integer)) and
                     not issubclass(tipo.type, (np.datetime64, np.timedelta64)))
-        return (isinstance(element, (float, int, np.floating, np.int_)) and
-                not isinstance(element, (bool, np.bool_, datetime, timedelta,
+        return (
+            isinstance(element, (float, int, np.floating, np.int_, np.long))
```

Comment:
I see you have addd `np.long` here. What do you think about adding a test for that case, exactly same as in original issue on Github?

Reply:
Done.